src/nodedef.cpp

```diff
@@ -207,7 +207,14 @@ void TileDef::serialize(std::ostream &os, u16 protocol_version) const
 	u8 version = 6;
 	writeU8(os, version);
 
-	os << serializeString16(name);
+	if (protocol_version > 39) {
+		os << serializeString16(name);
+	} else {
+		// Before f018737, TextureSource::getTextureAverageColor did not handle
+		// missing textures. "[png" can be used as base texture, but is not known
+		// on older clients. Hence use "blank.png" to avoid this problem.
+		os << serializeString16("blank.png^" + name);
```

Comment:
Doesn't this mean whatever texture the node has is scaled to the size of `blank.png`?

Also a workaround such as this must be avoided whenever possible, e.g. you can exclude all uses where `name[0] != '['`

Reply:
Done.

Reply:
> Doesn't this mean whatever texture the node has is scaled to the size of `blank.png`?

^ do you know?

Reply:
https://github.com/minetest/minetest/blob/8fab406c28bc5a18137d2653356c10880c827613/src/client/tile.cpp#L1165
https://github.com/minetest/minetest/blob/8fab406c28bc5a18137d2653356c10880c827613/src/client/tile.cpp#L1082-L1099

it upscales the smaller of both textures, whereas `blank.png` will always be the smaller one with a resolution of 1x1 px.